lxd/container_lxc.go

```diff
@@ -1369,8 +1373,26 @@ func (c *containerLXC) initLXC() error {
 	return nil
 }
 
+// Initialize the storage pool for this container.
+func (c *containerLXC) initStoragePool() error {
+	if c.storagePool != "" {
+		return nil
+	}
+
+	_, rootDiskDevice := containerGetRootDiskDevice(c.localDevices)
+	if rootDiskDevice["pool"] == "" {
+		_, rootDiskDevice = containerGetRootDiskDevice(c.expandedDevices)
+	}
+	if rootDiskDevice["pool"] == "" {
+		return fmt.Errorf("No storage pool found in the containers devices.")
+	}
```

Comment:
We shouldn't do that. Otherwise we run into the same problem I described before where if the container has a broken config, we can't ever fix it because all container functions call init() which calls initStoragePool() causing all of them to fail and requiring the user go fix the database by hand.